### PR TITLE
Add an ability to customize properties

### PR DESCRIPTION
**Pull request by pakrym**
You can change properties to fields, customize their names and accessibility.

cc @tg-msft @annelo-msft

### CODE REVIEW

File: test/TestProjects/TypeSchemaMapping/CustomizedModel.cs [hunk 1]
```diff
@@ -1,10 +1,15 @@
+using AnotherCustomNamespace;
 using Azure.Core;
 
 namespace CustomNamespace
 {
     [CodeGenSchema("Model")]
     internal partial class CustomizedModel
     {
+        [CodeGenSchemaMember("ModelProperty")]
```
Comment (pakrym) [inline test/TestProjects/TypeSchemaMapping/CustomizedModel.cs]: This is the sample usage

File: src/AutoRest.CSharp.V3/Output/Builders/BuilderHelpers.cs [hunk 2]
```diff
@@ -82,6 +82,17 @@ public static TypeDeclarationOptions CreateTypeAttributes(string defaultName, st
             );
         }
 
+        public static MemberDeclarationOptions CreatMemberDeclaration(string defaultName, CSharpType defaultType, Accessibility defaultAccessibility, ISymbol? existingMember)
+        {
+            // Not reading the return type information of existing members yet
```
Comment (tg-msft) [inline src/AutoRest.CSharp.V3/Output/Builders/BuilderHelpers.cs]: I could live without that for a long while.  Just being able to make one or two fields internal that I expose differently would be a big improvement.

File: src/AutoRest.CSharp.V3/Output/Models/Types/ObjectType.cs [hunk 3]
```diff
@@ -50,18 +55,62 @@ public ObjectType(ObjectSchema objectSchema, BuildContext context)
...
-        public CSharpType? ImplementsDictionaryElementType => _inheritsType ??= CreateInheritedDictionaryElementType();
+        public CSharpType? ImplementsDictionaryElementType => _implementsDictionaryType ??= CreateInheritedDictionaryElementType();
 
         public CSharpType Type { get; }
 
+        public ObjectTypeProperty GetPropertyForSchemaProperty(Property property, bool includeParents = false)
+        {
+            if (!TryGetPropertyForSchemaProperty(p => p.SchemaProperty == property, out ObjectTypeProperty? objectProperty, includeParents))
+            {
+                throw new InvalidOperationException($"Unable to find object property for schema property {property.SerializedName}");
```
Comment (tg-msft) [inline src/AutoRest.CSharp.V3/Output/Models/Types/ObjectType.cs]: Can we include the type in the error message?  In all the error messages you're adding, actually?  Unless SerializedName is fully qualified in which case ignore this.

File: src/AutoRest.CSharp.V3/Output/Models/Types/ObjectType.cs [hunk 4]
```diff
@@ -161,7 +206,13 @@ private ObjectTypeProperty CreateProperty(Property property)
             {
                 if (complexSchema is ObjectSchema parentObjectSchema)
                 {
-                    return _typeFactory.CreateType(parentObjectSchema, false);
+                    CSharpType type = _typeFactory.CreateType(parentObjectSchema, false);
+                    if (type.IsFrameworkType)
+                    {
+
+                    }
```
Comment (tg-msft) [inline src/AutoRest.CSharp.V3/Output/Models/Types/ObjectType.cs]: ?

File: src/AutoRest.CSharp.V3/Output/Builders/BuilderHelpers.cs [hunk 5]
```diff
@@ -82,6 +82,17 @@ public static TypeDeclarationOptions CreateTypeAttributes(string defaultName, st
             );
         }
 
+        public static MemberDeclarationOptions CreatMemberDeclaration(string defaultName, CSharpType defaultType, Accessibility defaultAccessibility, ISymbol? existingMember)
```
Comment (MiYanni) [inline src/AutoRest.CSharp.V3/Output/Builders/BuilderHelpers.cs]: Missing an 'e' in Create.

File: src/AutoRest.CSharp.V3/Generation/Writers/ModelWriter.cs [hunk 6]
```diff
@@ -62,10 +62,15 @@ private void WriteObjectSchema(CodeWriter writer, ObjectType schema)
...
+                        {
+                            continue;
+                        }
+
                         writer.WriteXmlDocumentationSummary(property.Description);
 
-                        CSharpType propertyType = property.Type;
-                        writer.Append($"public {propertyType} {property.Name:D}");
+                        CSharpType propertyType = property.DeclarationOptions.Type;
+                        writer.Append($"{property.DeclarationOptions.Accessibility} {propertyType} {property.DeclarationOptions.Name:D}");
```
[the remark this answers is not in the document]
Comment (pakrym) [reply src/AutoRest.CSharp.V3/Generation/Writers/ModelWriter.cs]: Maybe when we have more usages.

**Pull request merged**